poc-data-api/Extracting_Water_Temperature_at_Site.Rmd

```diff
@@ -0,0 +1,251 @@
+
+```{r explore}
+#Checking data structure
+data_df$schema
+```
+We can see that there are a number of variables available in this dataset. We will need to access three variables to create a list of all sampled sites in this monitoring program:  
+- `site`, which includes the name of all sites where temperature loggers were deployed
+- `subsite`, which includes a shorten version of the site name and an indication of where loggers where deployed (see below for more details)
+- `geometry`, which includes latitude and longitude coordinates in [well-known binary (WKB)](https://loc.gov/preservation/digital/formats/fdd/fdd000549.shtml) format
+  
+Based on the `subsite` information, we will create a new column that we will call `deployment_location`. If `subsite` string includes `SL`, this indicates the temperature logger was deployed on a reef slope (depths of 3 m or more). If this is the case, our new column will be labelled `reef slope`. If `FL` is included in the `subsite` string, this means the temperature logger was deployed on a reef flat (areas less than 3 m deep). In this case, we will classify this as `reef flat` in our new column.  
+  
+Next, we will transform the information contained in the `geometry` field into coordinate pairs (i.e., latitude and longitude in decimal degrees).  
+  
+**Note** that you can get additional information about the AIMS dataset by exploring its metadata as shown below.
+
+```{r metadata}
+#Checking metadata
+data_df$metadata
+```
+# Extracting variables of our interest from dataset
+
+We can extract data from the AIMS dataset by using `dplyr` verbs as shown below.
+
+```{r sites_bin}
+sites <- data_df %>% 
+  #We select unique sites included in the dataset
+  distinct(site, subsite, geometry) %>%
+  #This will load them into memory
+  collect()
+
+#Checking results
+glimpse(sites)
+```
+## Creating `deployment_location` column
+
+We will use the `subsite` column to create the categories in our new column, as explained in the [exploring the dataset structure](#exploring-dataset-structure). If no condition is met, then we will label the row as `NA`.
+
+```{r dep_loc}
+sites <- sites %>% 
+  #Adding new column - Given categories based on a condition
+  mutate(deployment_location =  case_when(str_detect(subsite, "FL[0-9]{1}") ~ "reef flat",
+                                          str_detect(subsite, "SL[0-9]{1}") ~ "reef slope",
+                                          #If no condition is met, return NA
+                                          T ~ NA))
+
+#Checking results
+glimpse(sites)
+```
+## Extracting latitude and longitude values (Option #1)
+As explained [above](#exploring-dataset-structure), the `geometry` field is in WKB format, which we will transform into degree coordinates in the next step. Here, we will use the `WKB` to transform the `geometry` and then we will convert it to an `sf` object for easy data manipulation.

```

Comment:
we will use the `readWKB` function from `WKB` package to transform the `geometry`